gcc/rust/backend/rust-compile-var-decl.h

```diff
@@ -78,15 +81,16 @@ class CompileVarDecl : public HIRCompileBase, public HIR::HIRPatternVisitor
   void visit (HIR::TupleStructPattern &) override {}
 
 private:
-  CompileVarDecl (Context *ctx, tree fndecl, tree translated_type)
+  CompileVarDecl (Context *ctx, tree fndecl, tree translated_type,
+		  std::vector<Bvariable *> *locals)
     : HIRCompileBase (ctx), fndecl (fndecl), translated_type (translated_type),
-      compiled_variable (ctx->get_backend ()->error_variable ())
+      locals (locals)
   {}
 
   tree fndecl;
   tree translated_type;
 
-  Bvariable *compiled_variable;
+  std::vector<Bvariable *> *locals;
```

Comment:
And then keep a reference here instead :)
```suggestion
  std::vector<Bvariable *> &locals;
```